DocsScraper/src/DocsScraper.jl

```diff
@@ -0,0 +1,152 @@
+"""
+    get_base_url(url::String)
+
+Extracts the base url.
+
+# Arguments
+- `url`: The url string of which, the base url needs to be extracted
+"""
+function get_base_url(url::String)
+    parsed_url = URIs.URI(url)
+    base_url = string(parsed_url.scheme, "://", parsed_url.host, parsed_url.port != nothing ? ":" * string(parsed_url.port) : "", parsed_url.path)
+    return base_url
+end
+
+
+"""
+    process_nodes(node, heading_hierarchy::Dict, parsed_text::Vector)
+
+Process the nodes recursively. The function recursively calls itself for every HTML hierarchy, thereby going deeper to retrieve the text.
+
+# Arguments
+- `node`: The root HTML node 
+- `heading_hierarchy`: Dictionary used to store metadata
+- `parsed_text`: Vector of Dicts to store parsed text and metadata
+
+# TODO:
+- Inline code blocks are the ones present inside <p> or <li>. put `` around the inline code blocks. Make changes inside `elseif tag_name == "p" || tag_name == "li"`
+"""
+function process_nodes(node, heading_hierarchy::Dict, parsed_text::Vector)
+    # if a node is an html element
+    if typeof(node) <: Gumbo.HTMLElement
+
+        tag_name = String(Gumbo.tag(node))
+
+        # process headings. Working: If the current tag is a heading, then, we remove all the smaller headings we stored (for metadata) from  
+        # heading_hierarchy dictionary and add the current heading tag
+        # This is done because, in the heading hierarchy, if we come back to the current heading, 
+        # then there is no need for the smaller ehadings we encountered earlier
+        if startswith(tag_name, "h") && isdigit(last(tag_name))
+            # Clear headings of equal or lower level
+            for k in collect(keys(heading_hierarchy))
+                if Base.parse(Int, last(k)) >= Base.parse(Int, last(tag_name))
+                    delete!(heading_hierarchy, k)
+                end
+            end
+            heading_hierarchy[tag_name] = strip(Gumbo.text(node))
+            push!(parsed_text, Dict("heading" => strip(Gumbo.text(node)), "metadata" => copy(heading_hierarchy)))
+
+            # if the current tag is <code>, then get it's textual value and store in the dictionary with key as "code"
+        elseif tag_name == "code"
+            # Start a new code block
+            code_content = strip(Gumbo.text(node))
+            push!(parsed_text, Dict("code" => code_content, "metadata" => copy(heading_hierarchy)))
+            return ""
+
+            # if the current tag is <p> or <li>, then store the text of the whole tag 
+        elseif tag_name == "p" || tag_name == "li"
```

Comment:
Yes, there are other tags like span, ul, ol, table, etc., but if any other tag contains text, then it'll eventually end up inside HTMLText (since we're going deeper into html hierarchy in every iteration), so there's no need to explicitly handle the others